substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/localization/OptimizedLocalizationSupport.java

```diff
@@ -76,6 +80,32 @@ public ResourceBundle getCached(String baseName, Locale locale) throws MissingRe
 
     }
 
+    private final Field bundleNameField = ReflectionUtil.lookupField(ResourceBundle.class, "name");
+    private final Field bundleLocaleField = ReflectionUtil.lookupField(ResourceBundle.class, "locale");
+
+    @Override
+    public void prepareClassResourceBundle(String basename, Class<?> bundleClass) {
+        try {
+            ResourceBundle bundle = ((ResourceBundle) ReflectionUtil.newInstance(bundleClass));
+            Locale locale = extractLocale(bundleClass);
+            /*- Set the basename and locale to be consistent with JVM lookup process */
+            bundleNameField.set(bundle, basename);
+            bundleLocaleField.set(bundle, locale);
```

Comment:
As far as I know, it should not. There is no special logic in their getters and setters and they are used only during the lookup process in the ResourceBundle class, which is completely substituted in the optimized mode. There is no suspicious use outside the class either. I am setting them reflectively because they are also set during the lookup, see [ResourceBundle.loadLookup](https://github.com/graalvm/labs-openjdk-11/blob/master/src/java.base/share/classes/java/util/ResourceBundle.java#L2006). Without setting them, the class bundles in the optimized mode would have no locale and no basename associated, which would be inconsistent with the JDK. The whole idea of not instantiating bundles and only preparing their classes for reflection, unfortunately, does not work with the older optimized mode, in which everything is stored in the map at build time, so I have to instantiate them anyway.